### PR TITLE
Add jscs

### PR DESCRIPTION
**Pull request by Andersos**
I added the jscs file from https://github.com/finn-no/js-code-style/blob/master/.jscsrc
I removed the `"requireCamelCaseOrUpperCaseIdentifiers": true` check because of redundant check https://github.com/finn-no/js-code-style/issues/4
I updated the errors I found.
The tests are not running on the admin code.

What do you think? Should we add jscs?

### CODE REVIEW

**Comment by jarib** [comment]
Does it replace jshint or do we need both? What's the benefit of jscs?

File: package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
     "start": "NODE_ENV=production node server.js",
     "build": "./node_modules/.bin/webpack",
     "dev": "NODE_ENV=development supervisor --ignore ./node_modules/,./public/js server.js",
-    "test": "export PORT=4243 ; jest && jshint server.js lib test && jsxhint public/js/**/*.jsx && mocha test test/*.js && npm run coverage",
+    "test": "export PORT=4243 ; jshint server.js lib test && jscs server.js lib test && jsxhint public/js/**/*.jsx && jest && mocha test test/*.js && npm run coverage",
```
Comment (jarib) [inline package.json]: The test script is getting long. Maybe it'd be nicer to have it as a shell script in the `scripts/` folder?

**Comment by Andersos** [comment]
From my understanding we can use both. JSCS is coding style like us choosing to use single quotes. More info here http://jscs.info/

File: package.json (hunk 1 above)
Comment (Andersos) [inline package.json]: Agree :+1:

**Comment by jarib** [comment]
Isn't that exactly what JsHint gives us? Does JSCS support JSX? I'm all for enforcing code style rules, but if add a new tool there should be some clear benefit.

**Comment by Andersos** [comment]
Maybe @gregersrygg or @sveisvei have some opinions about using both JSCS and JSHint or the advantage of using JSCS.

**Comment by ivarconr** [comment]
@Andersos does JSCS bring more style control than what we can get from jshint?

**Comment by Andersos** [comment]
JSCS definitively adds more style control (see http://jscs.info/rules.html) but it seems they don't overlap a lot. I just thought about adding it because it was [added](https://github.com/finn-no/js-code-style/commit/a1cf667fffd10d8059baa7428a8c43310c113111) to the [finn-js-code-style](https://github.com/finn-no/js-code-style). I have not used JSCS before this.

**Comment by gregersrygg** [comment]
The jscs in finn-js-code-style is by no means official. We need to discuss and agree on the rules before projects start to use it (imho).
JSCS has a lot more details than jshint have for some things, but I haven't looked into it enough yet. There is also eslint, that is more pluggable than jshint and jscs. So that's a third alternative.

Just use finn-js-code-style (the cli). For now it only uses jshint, but the idea is that we can use multiple tools with only one command (finn-js-code-style) in the future.

**Pull request closed without merging**